### ISSUE TITLE
parser: `BindingIdentifier` `typeAnnotation` is typed as `null` in generated `types.d.ts`

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#22134]
The generated parser TypeScript types currently declare `BindingIdentifier.typeAnnotation` as always `null`, but the parser can emit a real `TSTypeAnnotation` for binding identifiers.

Reference discussion: https://discord.com/channels/1079625926024900739/1080712024683708466/1500159726980432084

AI disclosure: this issue text was drafted with AI assistance and reviewed before filing.

### Reproduction

Parse a TypeScript binding identifier with a type annotation, for example:

```ts
const x: string = "foo";
```

The parsed AST includes a `BindingIdentifier` whose `typeAnnotation` is populated:

```ts
{
  type: "Identifier",
  name: "x",
  typeAnnotation: {
    type: "TSTypeAnnotation",
    typeAnnotation: {
      type: "TSStringKeyword"
    }
  }
}
```

### Current Type Definition

In `types.d.ts`, `BindingIdentifier` is currently generated as:

```ts
export interface BindingIdentifier extends Span {
  type: "Identifier";
  decorators?: [];
  name: string;
  optional?: false;
  typeAnnotation?: null;
  parent?: Node;
}
```

### Expected

`IdentifierReference.typeAnnotation` should remain `null`, because annotations are not valid on identifier references:

```ts
const y = x: string; // invalid
```

But `BindingIdentifier.typeAnnotation` should allow a real TypeScript annotation, because bindings can be annotated:

```ts
const x: string = "foo";
```

Expected type:

```ts
export interface BindingIdentifier extends Span {
  type: "Identifier";
  decorators?: [];
  name: string;
  optional?: false;
  typeAnnotation?: TSTypeAnnotation | null;
  parent?: Node;
}
```

### Notes

This appears to be a generated type mismatch rather than a parser AST output issue. The generated files point to:

```ts
tasks/ast_tools/src/generators/typescript.rs
```

as the source for `types.d.ts` generation.

**Comment by parkrevil** [comment]
Thanks for the detailed write-up, @camc314 — the analysis here (including the pointer to `tasks/ast_tools/src/generators/typescript.rs`) matches what we ran into downstream. Adding a bit of extra confirmation in case it's useful for prioritising.

**Still reproduces on the latest release (`0.133.0`)**
`@oxc-project/types@0.133.0` still declares:
```ts
// BindingIdentifier
typeAnnotation?: null;
```
while `oxc-parser@0.133.0` populates a real `TSTypeAnnotation` at runtime:
```js
parseSync("t.ts", "function f(a: string) {}").program.body[0].params[0]
//   .typeAnnotation.typeAnnotation.type === "TSStringKeyword"
```

**Cross-check against the format the README references**
The README notes the TS AST conforms to `typescript-estree`. For reference, the typescript-eslint type definitions declare the annotation on the identifier node as:
```ts
// @typescript-eslint/types — Identifier
typeAnnotation: TSTypeAnnotation | undefined;
```
so the proposed `typeAnnotation?: TSTypeAnnotation | null` lines up with the format the parser targets. (The neighbouring `decorators?: []` field looks like it may be the same generated-type gap.)

**Downstream impact**
Consumers reading parameter/variable type annotations currently have to cast away `null` even though the runtime data is correct.

Totally understand this may be low priority. If it'd help, I'm happy to open a PR against the generator — could a maintainer confirm the intended shape, including which identifier nodes should be widened vs. left as `null`? Thanks again for maintaining oxc!

_(Drafted with AI assistance and reviewed before posting.)_
